slides/src/content/unit-testing.md

```diff
@@ -128,117 +141,87 @@ describe('CapitalizePipe', () => {
 ```
 
 ---
-<!-- .slide: id="unit-testing-getting-tests-to-run" -->
-## Getting Tests to Run
+<!-- .slide: id="unit-testing-instantiating-component--or-service" -->
+## Instantiating the Component or Service
 
-- Run `ng test`: 4 failures
-- Have to include our dependencies in `app.component.spec.ts`
-  - And in `to-do-list.component.spec.ts`
+- We currently do not display the number of todos in `AppComponent`.
+- Let's create a method that will return the number of todos.
+- Start by removing existing test code until we have the shell of a simple test for this component.
 
 #####_src/app/app.component.spec.ts_
 ```ts
-    TestBed.configureTestingModule({
-      declarations: [
-        AppComponent,
-        ToDoListComponent,
-        GenericInputComponent
-      ],
-      providers: [
-        ToDoService
-      ]
-    }).compileComponents();
-```
+import { AppComponent } from './app.component';
 
----
-<!-- .slide: id="unit-testing-cleaning-up-tech-debt" -->
-## Cleaning Up Technical Debt
-
-- Title is now "To Do" instead of "app works!"
-- Notice we're no longer passing data to the display in `thingsToDo`,
-  so remove it from the HTML and erase the `@Input`
+fdescribe('AppComponent', () => {
 
-![Auto-Generated Tests Running](content/images/screenshot-jasmine-defaults-run.png)
+  beforeEach(() => {
 
----
-<!-- .slide: id="unit-testing-instantiate-component" -->
-## Instantiating the Component or Service
+  });
 
-- How would we test `itemCount` on `AppComponent`?
+  it('should count the number of items', () => {
 
-#####_src/app/app.component.ts_
-```ts
-export class AppComponent {
-  // ...as before...
-  itemCount() {
-    return this.toDoService.itemCount();
-  }
-}
-```
+  });
 
-#####_src/app/to-do.service.ts_
-```ts
-export class ToDoService {
-  // ...as before...
-  itemCount() {
-    return this.items.length;
-  }
-}
+});
 ```
 
 ---
 <!-- .slide: id="unit-testing-mocking-dependencies" -->
 ## Mocking Dependencies
 
 - We need an object that matches the "shape" of `TodoService` to test `AppComponent`
-- The mock object must have `onNewItem` and `itemCount` methods
-- `onNewItem` doesn't actually need to do anything
-- `itemCount` can return a fixed value
+- Note the use of `as` to assert the type of our mock.
```

Comment:
This is just an assertion for static type checking and has no impact on runtime code. Rather than changing the type, it's telling TypeScript to *act* like it's the other type.